File: backend/src/main/java/turip/favorite/controller/dto/response/FavoriteFolderResponse.java
package turip.favorite.controller.dto.response;

import turip.favorite.domain.FavoriteFolder;

public record FavoriteFolderResponse(
        Long id,
        Long memberId,
        String name,
        boolean isDefault
) {

    public static FavoriteFolderResponse from(FavoriteFolder favoriteFolder) {
        return new FavoriteFolderResponse(favoriteFolder.getId(), favoriteFolder.getMember().getId(),
                favoriteFolder.getName(), favoriteFolder.isDefault()
        );
    }
}

File: android/app/src/main/java/com/on/turip/data/folder/dto/FavoriteFolderResponse.kt
package com.on.turip.data.folder.dto

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class FavoriteFolderResponse(
    @SerialName("id")
    val id: Long,
    @SerialName("isDefault")
    val isDefault: Boolean,
    @SerialName("memberId")
    val memberId: Long,
    @SerialName("name")
    val name: String,
    @SerialName("placeCount")
    val placeCount: Int,
)
